python_modules/dagster/dagster/_time/__init__.py

```diff
@@ -0,0 +1,135 @@
+import time
+from datetime import datetime, timedelta, timezone, tzinfo
+from typing import Union
+
+from dateutil import parser
+
+import dagster._check as check
+
+try:
+    # zoneinfo is python >= 3.9
+    from zoneinfo import ZoneInfo as _timezone_from_string  # type: ignore
+except:
+    from dateutil.tz import gettz as _timezone_from_string
+
+
+def _mockable_get_current_datetime() -> datetime:
+    # Can be mocked in tests by freeze_time()
+    return datetime.now(tz=timezone.utc)
+
+
+def get_current_datetime(tz="UTC") -> datetime:
+    """Return the current datetime. Will always have a timezone
+    (defaults to UTC if none is specified). Value can be mocked
+    via dagster._core.test_utils.freeze_time.
+    """
+    utc_datetime = _mockable_get_current_datetime()
+
+    if tz == "utc" or tz == "UTC":
+        return utc_datetime
+
+    return utc_datetime.astimezone(get_timezone(tz))
+
+
+def _mockable_get_current_timestamp() -> float:
+    return time.time()
+
+
+def get_current_timestamp() -> float:
+    """Return the current unix timestamp. Value can be mocked
+    via dagster._core.test_utils.freeze_time.
+    """
+    # Like time.time() but can be mocked in tests by freeze_time()
+    return _mockable_get_current_timestamp()
+
+
+def get_timezone(timezone_name: str) -> tzinfo:
+    """Creates a tzinfo object with the given IANA timezone name."""
+    if timezone_name == "utc" or timezone_name == "UTC":
+        return timezone.utc
+
+    return check.not_none(_timezone_from_string(timezone_name))
+
+
+def create_datetime(year, month, day, *args, **kwargs):
+    """Creates a datetime object. Same arguments as datetime.datetime
+    constructor, but will always have a timezone (defaults to UTC if none is specified).
+    """
+    tz = kwargs.pop("tz", "UTC")
+    if isinstance(tz, str):
+        tz = get_timezone(tz)
+    return datetime(year, month, day, *args, **kwargs, tzinfo=tz)
+
+
+def datetime_from_timestamp(timestamp: float, tz: Union[str, tzinfo] = timezone.utc) -> datetime:
+    """Creates a datetime object from a unix timestamp. Will always have a timezone
+    (defaults to UTC if none is specified).
+    """
+    if not tz:
+        tzinfo = timezone.utc
+    elif isinstance(tz, str):
+        tzinfo = get_timezone(tz)
+    else:
+        tzinfo = tz
+
+    return datetime.fromtimestamp(timestamp, tz=tzinfo)
+
+
+def add_fixed_time(
```

Comment:
I don't feel strongly but if we're considering naming here are some other ideas:
- add_absolute_time
- add_timezone_agnostic
- add_time_in_utc

Reply:
`progress_time` 
`safe_add_time` 
`update_dt`

Reply:
I went with add_absoltuie_time